components/script/script_thread.rs

```diff
@@ -1526,6 +1535,33 @@ impl ScriptThread {
         document.run_the_animation_frame_callbacks();
     }
 
+    /// Handles firing of transition events.
+    #[allow(unsafe_code)]
+    fn handle_transition_event(&self, unsafe_node: UnsafeNode, name: String, duration: f64) {
+        let node = unsafe { ServoLayoutNode::from_unsafe(&unsafe_node) };
+        let node = unsafe { node.get_jsmanaged().get_for_script() };
+
+        if let Some(el) = node.downcast::<Element>() {
+            if &*window_from_node(node).GetComputedStyle(el, None).Display() == "none" {
+                return;
+            }
+        }
+
+        let init = TransitionEventInit {
+            parent: EventInit {
+                bubbles: true,
+                cancelable: false,
+            },
+            propertyName: DOMString::from(name),
+            elapsedTime: Finite::new(duration as f32).unwrap(),
+            pseudoElement: DOMString::new()
```

Comment:
I wanted to, but I'm not entirely sure how to. How can you extract information about whether a transition is being called upon a pseudo-element instead of an element if you have the element node itself?